psycopg/psycopg/cursor_async.py

```diff
@@ -138,23 +138,32 @@ async def stream(
         params: Optional[Params] = None,
         *,
         binary: Optional[bool] = None,
+        size: int = 1,
     ) -> AsyncIterator[Row]:
         """
         Iterate row-by-row on a result from the database.
+
+        :param size: if greater than 1, results will be retrieved by chunks of
+            this size from the server (but still yielded row-by-row); this is only
+            available from version 17 of the libpq.
         """
         if self._pgconn.pipeline_status:
             raise e.ProgrammingError("stream() cannot be used in pipeline mode")
 
         async with self._conn.lock:
             try:
                 await self._conn.wait(
-                    self._stream_send_gen(query, params, binary=binary)
+                    self._stream_send_gen(query, params, binary=binary, size=size)
                 )
                 first = True
                 while await self._conn.wait(self._stream_fetchone_gen(first)):
-                    # We know that, if we got a result, it has a single row.
-                    rec: Row = self._tx.load_row(0, self._make_row)  # type: ignore
-                    yield rec
+                    # We know that, if we got a result, it has a single row or
+                    # a chunk of rows.
```

Comment:
I think we can drop this comment altogether. The previous one was about assumption about there being exactly one record, but here we are making a careful iteration on >= 0 records.
```suggestion
```